src/instana/collector/helpers/runtime.py

```diff
@@ -10,29 +11,92 @@
+from typing import Any, Dict, List, Union, Callable
 
 from instana.collector.helpers.base import BaseHelper
 from instana.log import logger
 from instana.util import DictionaryOfStan
 from instana.util.runtime import determine_service_name
 from instana.version import VERSION
+from instana.collector.base import BaseCollector
 
 PATH_OF_DEPRECATED_INSTALLATION_VIA_HOST_AGENT = "/tmp/.instana/python"
 
-PATH_OF_AUTOTRACE_WEBHOOK_SITEDIR = '/opt/instana/instrumentation/python/'
+PATH_OF_AUTOTRACE_WEBHOOK_SITEDIR = "/opt/instana/instrumentation/python/"
+
 
-def is_autowrapt_instrumented():
-    return 'instana' in os.environ.get('AUTOWRAPT_BOOTSTRAP', ())
+def is_autowrapt_instrumented() -> bool:
+    return "instana" in os.environ.get("AUTOWRAPT_BOOTSTRAP", ())
 
 
-def is_webhook_instrumented():
+def is_webhook_instrumented() -> bool:
     return any(map(lambda p: PATH_OF_AUTOTRACE_WEBHOOK_SITEDIR in p, sys.path))
 
 
+def parse_ignored_endpoints(
+    params: Union[Dict[str, Any], str],
+) -> List[str]:
+    """
+    This function parses input to prepare a list for ignored endpoints.
+
+    @param params: Can be either:
+        - String: "service1:endpoint1,endpoint2;service2:endpoint3" or "service1;service2"
+        - Dict: {"service1": ["endpoint1", "endpoint2"], "service2": ["endpoint3"]}
+    @return: List of strings in format ["service1.endpoint1", "service1.endpoint2", "service2"]
+    """
+    try:
+        ignore_endpoints = []
+
+        # If set from env variable (string format)
+        if isinstance(params, str):
+            if not params:
+                return ignore_endpoints
+
+            instrumentation_pairs = params.split(";")
+
+            for pair in instrumentation_pairs:
+                instrumentation = pair.strip()
+
+                if not instrumentation:
+                    continue
+
+                if ":" in pair:
+                    instrumentation, endpoints = pair.split(":", 1)
+                    instrumentation = instrumentation.strip()
+
+                    # Split endpoints by comma and clean whitespace
+                    endpoint_list = [
+                        ep.strip() for ep in endpoints.split(",") if ep.strip()
+                    ]
+                    if endpoint_list:
+                        for endpoint in endpoint_list:
+                            ignore_endpoints.append(f"{instrumentation}.{endpoint}")
+                else:
+                    # Handle case where only service name is provided
+                    ignore_endpoints.append(instrumentation)
+
+        # If set from agent configuration (dict format)
+        elif isinstance(params, dict):
+            for service, endpoints in params.items():
+                if not endpoints:  # If endpoints list is empty
+                    ignore_endpoints.append(service)
+                else:
+                    for endpoint in endpoints:
+                        ignore_endpoints.append(f"{service}.{endpoint}")
```

Comment:
Let's practice the `Clean Code` here. It would be better to split this whole function in two: one to parse string ignored endpoints and another to parse dictionary ignored endpoints.